### DISCUSSION TITLE
GitVersion tool is taking a long time to generate the version

### DISCUSSION THREAD

**Discussion by suman-dinda** [GitTools/GitVersion#4013] [Q&A]
Hello everyone, 

I'm currently running the Azure DevOps GitTools official task to generate the semantic version (semver). However, it's taking quite a long time, approximately 12 minutes, which is significantly impacting the overall build time of our code. 

I'm seeking suggestions on how we can optimize this process to reduce its duration, or if there are any alternative approaches we could explore. 

Thank you in advance for your assistance.

**Comment by thomassvensen** [comment]
Hi

We had the same issue, and decided to upgrade to https://github.com/GitTools/GitVersion/releases/tag/6.0.0-beta.5 as a temporary fix. It reduce running time from minutes to seconds. We needed to change `GitVersion.yml` from using `tag:` to `label:`, but other then that it was a drop-in replacement for us. It has been running stable for a couple of months now, so has been a good choice for us at least.

Regards
Thomas

**Comment by HHobeck** [comment]
Hi @suman-dinda,

It would be good to provide any information about your environment and how your branching and merging strategy is.

May I ask you to help with the following one or more points:

- Give background information about your context/environment
- Find out where the problem might be located (Is it because of so many tags? Is it because of so many branches? Is it because you are using a special mode?)
- Create a repository which can be used for analyzing and improving the performance
- Statistic how the complexity is in different circumstances (https://en.wikipedia.org/wiki/Time_complexity)

BTW: There is already a pre-release (see version 6.x or main branch) existing which includes a lot of changes. At this moment it would make no sense to invest time on version 5.x.
